### DISCUSSION TITLE
Expose the Custom button renderer as a context instead of a render function approach

### DISCUSSION THREAD

**Discussion by GuiBibeau** [rainbow-me/rainbowkit#538] [Ideas]
Hello!

I really think your nailed the UX on this library! I would love to see the Custom button however be a react context so that we could have a cleaner component hierarchy when implementing a custom button.

Right now the renderer works fine but makes it hard to suspend components while unmounted or to split properly into sub components. It is totally possible but the API to consume is a little be convoluted inside a function.

Exposing the exact same data in context with getter hooks would be a pleaser to work with! I'm open to contributing if the suggestion is accepted.

**Comment by markdalgleish** [comment]
You might find it helpful that we just shipped Hooks for opening the modals in [RainbowKit v0.4.3.](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.4.3) You can check out the docs here: https://www.rainbowkit.com/docs/modal-hooks

If you use these along with wagmi Hooks, you should be able to build your own custom connect button without needing `ConnectButton.Custom`. If anything's missing or difficult, please feel free to raise another discussion with your specific issue as I'm interested in exploring higher-level Hooks.

**Reply by GuiBibeau** [reply]
This is golden! Thank you very much!
